readme.md

```diff
@@ -44,7 +44,7 @@ Yields:
 
 ## API
 
-### visit([node](https://github.com/wooorm/unist#unist-nodes)\[, type\], callback\[, reverse\])
+### visit([node](https://github.com/wooorm/unist#unist-nodes)\[, type], callback\[, reverse])
```

Comment:
> This is what I meant by "unsustainable"

I’m not sure it’s unsustainable, people have been relying on this a lot already using other markdown processors:

```md
[foo]
```

Or more commonly in something like this:

```md
# remark.parse(doc[, options])
```

Do we really need to escape those just in case someone adds a definition for those references later?

```md
[foo]

[foo]: http://example.com
```

Or can we ask users to escape themselves where needed:


```md
\[foo]

[foo]: http://example.com
```

Resulting in a `text` node with `value: '[foo]'`, and a definition with an identifier of `foo`, in which case we do escape `[foo]` when compiling?